src/python/pants/engine/internals/build_files.py

```diff
@@ -235,6 +235,7 @@ async def _extract_env_vars(
 @rule(desc="Search for addresses in BUILD files")
 async def parse_address_family(
     parser: Parser,
+    bootstrap_status: BootstrapStatus,
```

Comment:
I guess this will cause reevaluation for all BUILD files that were parsed during bootstrap, hopefully that is a limited number though..

Reply:
Yea, it should be. Bootstrap should only evaluate files containing `environment` targets, and their parents.